packages/launch/src/hooks/use-domain-search.ts

```diff
@@ -10,30 +10,38 @@ import { isGoodDefaultDomainQuery } from '@automattic/domain-picker';
  */
 import { LAUNCH_STORE } from '../stores';
 import { useSite, useTitle } from './';
-import { isDefaultSiteTitle } from '../utils';
 import { useSiteDomains } from './use-site-domains';
 
 export function useDomainSearch(): {
 	domainSearch: string;
 	isLoading: boolean;
 	setDomainSearch: ( search: string ) => void;
 } {
-	const { domainSearch } = useSelect( ( select ) => select( LAUNCH_STORE ).getState() );
-	const { title } = useTitle();
-	const { isLoadingSite } = useSite();
+	const existingDomainSearch = useSelect(
+		( select ) => select( LAUNCH_STORE ).getDomainSearch(),
+		[]
+	);
+	const { isDefaultTitle, title } = useTitle();
+	const { isLoadingSite: isLoading } = useSite();
 	const { siteSubdomain } = useSiteDomains();
 
 	const { setDomainSearch } = useDispatch( LAUNCH_STORE );
 
-	let search = domainSearch.trim() || filterUnsuitableTitles( title || '' );
+	const nonDefaultTitle = ( title && title.length > 1 && ! isDefaultTitle && title ) || '';
```

Comment:
A more explicit way of writing this check could be:

```suggestion
	const nonDefaultTitle = typeof title !== 'undefined' && title.length > 1 && ! isDefaultTitle ? title : '';
```

Advantages:
- Explicit typeguard against `undefined` for `title` 
- Use of ternary operator — which makes the logic a bit more clear